core/template.yaml

```
@@ -149,6 +158,13 @@ Resources:
         ServerSideEncryptionConfiguration:
           - ServerSideEncryptionByDefault:
               SSEAlgorithm: AES256
+      ObjectLockEnabled: true
+      ObjectLockConfiguration:
+        ObjectLockEnabled: Enabled
+        Rule:
+          DefaultRetention:
+            Mode: !If [ IsProd, "COMPLIANCE", "GOVERNANCE" ]
+            Days: 1
```

Are we able to make this optional in dev?
Something like....
```suggestion
      ObjectLockConfiguration:
        ObjectLockEnabled: !If [IsDev, false, true]
        Rule:
          DefaultRetention:
            Mode: !If [ IsProd, "COMPLIANCE", "GOVERNANCE" ]
            Days: 1
```
I removed the separate properly `ObjectLockEnabled` because I'm not sure if it's required, but if it is you can add the same expression

---

I think it’s a bit tricky because the first `ObjectLockEnabled` parameter (which can be either `true`/`false`) for the bucket is required if `ObjectLockConfiguration` is present, otherwise the stack gives this error:

`Resource handler returned message: "Object lock configuration can only be set on Object lock enabled buckets" (HandlerErrorCode: InvalidRequest)`

If it’s set to `true`, then the `ObjectLockConfiguration` is used as expected, but if it’s set to `false`, and `ObjectLockConfiguration` is present, we get the same error as above.

Also, the second `ObjectLockEnabled` parameter under the `ObjectLockConfiguration` can only take the single value of `Enabled` as per [the docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-properties-s3-bucket-objectlockconfiguration.html) (empty string/null gives an error). I also tried a conditional to set the retention period to zero in dev, but it has to be a positive integer

This only thing I can think of is having something like !`If [IsDev, 1, 30]`?

---

_Interesting_! 
You _can_ make the resource itself conditional! So have an object lock enabled resource definition for prod-like environments, and a non-object-lock enabled resource for everything else?

Feel free to resolve this and merge it in. I'm just trying to help manage the amount of complexity in the system.